tools/generator/files/api.hh.jinja2

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+// Copyright (c) 2012-2020 Association Prologin <association@prologin.org>
+
+#pragma once
+
+#include <memory>
+{#- The following two headers may not be necessary. They are only used if there
+is a std::string, or a std::vector in the generated function signatures.
+We could make the test here and add the includes only if necessary, but it does
+not cost must to include them in every cases. #}
+#include <string>
+#include <vector>
+
+#include <rules/api.hh>
+#include <rules/player.hh>
+
+#include "actions.hh"
+#include "constant.hh"
+#include "game_state.hh"
+
+/**
+ * The methods of this class are exported through 'interface.cc'
+ * to be called by the clients
+ */
+class Api final : public rules::Api<GameState, error>
+{
+public:
+    Api(std::unique_ptr<GameState> game_state, rules::Player_sptr player);
+    ~Api() {}
+
+{#- Generate the C++ declaration for each API function -#}
+{% for func in yaml.function %}
+{# Since this is a header, generate the function documentation #}
+    {{ ("/// " + func.fct_summary)|wordwrap(79, False, "\n    /// ") }}
```

Comment:
Maybe this could deserve its own filter? `func|cxx_comment` with a `doc=False` argument for `///`?

Reply:
I’m not sure about this. Here it is really a job for the template engine, since it is just doing a wordwrap.

Also, I did not understood the `doc=False`